app/src/androidTest/java/com/example/avjindersinghsekhon/minimaltodo/testApp/tests/Helpers.java

```diff
@@ -1,4 +1,124 @@
-}
+
+    public static String getText(final Matcher<View> matcher) {
+        final String[] stringHolder = {null};
+        onView(matcher).perform(new ViewAction() {
+            @Override
+            public Matcher<View> getConstraints() {
+                return isAssignableFrom(TextView.class);
+            }
+
+            @Override
+            public String getDescription() {
+                return "getting text from a TextView";
+            }
+
+            @Override
+            public void perform(UiController uiController, View view) {
+                TextView tv = (TextView) view; //Save, because of check in getConstraints()
+                stringHolder[0] = tv.getText().toString();
+            }
+        });
+        return stringHolder[0];
+    }
+
+    public static String substring(String str) {
+        String input = str.substring(16, 29);
+        SimpleDateFormat parser = new SimpleDateFormat("dd MMM,yyyy");
+        Date date = null;
+        try {
+            date = parser.parse(input);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        SimpleDateFormat formatter = new SimpleDateFormat("dd MMM, yyyy");
+        return formatter.format(date);
+    }
+
+    public static void clickEl(Matcher<View> matcher) {
+        onView(matcher).perform(click());
+    }
+
+    public static boolean isCheckDisplay(Matcher<View> matcher) {
+        try {
+            onView(matcher).check(matches(isDisplayed()));
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    public static boolean isTextAbove(Matcher<View> matcher, Matcher<View> matcher2) {
+        try {
+            onView(matcher).check(isAbove(matcher2));
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    public static boolean isElemClickable(Matcher<View> matcher) {
+        try {
+            onView(matcher).check(matches(isClickable()));
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    public static boolean isTextLeft(Matcher<View> matcher, Matcher<View> matcher2) {
+        try {
+            onView(matcher).check(isLeftOf(matcher2));
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    public static boolean isTextRight(Matcher<View> matcher, Matcher<View> matcher2) {
+        try {
+            onView(matcher).check(isRightOf(matcher2));
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    public static boolean isTextBelow(Matcher<View> matcher, Matcher<View> matcher2) {
+        try {
+            onView(matcher).check(isBelow(matcher2));
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    public static void performUppercase(Matcher<View> matcher, String args) {
```

Comment:
Include the action of "typeText" in this name

Reply:
Add a better name for "args"

Reply:
Done